Code-the-Dream-School/cc-prac-team4-front
# PP-89: Add functionality to User Update Profile page. Formatting the …

Add functionality to User Update Profile page. Formatting the date string on Profile page implemented. Added logout function on Profile page.

**This pull request include the following updates:**
- Configured Axios for user actions: logout and updateUserProfile.
- Implemented the reducer function profileReducer to handle different actions.
- Added functionality to User Update Profile page [http://localhost:3000/me/update](http://localhost:3000/me/update):
> - Implemented formatting for the date string on the Profile page.
> - Added a logout function on the Profile page.
- Updates in the Profile component:
> - Modified the logout action handling.
> - Displayed the joined date with formatted and relative time.
- Updates in the UpdateProfile component:
> - Implemented form submission handling.
> - Displayed error and success messages.

**Testing instructions:**
Install:
- npm install
- npm install moment date-fns
- Pay special attention to the changes made in the Profile.js and UpdateProfile.js files.
- Test the updated profile functionality by submitting the form with valid or invalid data and verifying the success or error messages.
- Check if the profile information is updated correctly after submitting the form.
- Ensure that any error messages or success messages related to the current pull request are displayed correctly.
- Test the logout functionality by clicking the logout button and ensuring that the user is successfully signed out and redirected to the home page.

<img width="1011" alt="User Update Profile Page" src="https://github.com/AlonaVladymyrovaTrinity/adopt-a-pet/blob/main/UserUpdateProfilepage.png?raw=true">
